libpages/server.go

```diff
@@ -143,21 +150,79 @@ func (s *Server) isDirWithNoIndexHTML(
 	}
 }
 
+const cloningFilename = "CLONING"
+const gitRootInitialTimeout = time.Second
+
+func (s *Server) shouldShowCloningLandingPage(st *site) (bool, error) {
+	if st.root.Type != GitRoot {
+		// CLONING file only matters for git roots.
+		return false, nil
+	}
+	newCtx, err := libkbfs.NewContextWithCancellationDelayer(
+		libkbfs.CtxWithRandomIDReplayable(context.Background(),
+			ctxIDKey, ctxOpID, nil))
```

Comment:
This will leak a goroutine. `libkbfs.NewContextWithCancellationDelayer` must take a cancellable context, which itself is at some point canceled. I'd suggest you make the below timeout context first (and `defer` its `cancel`), and then pass that into the `CtxWithRandomIDReplayable` above.

Reply:
Good catch; I'm realizing this is an issue in MakeFS as well. Will fix that too. Thanks